golpa.go

```diff
@@ -404,3 +407,45 @@ func (model *Model) SolveWithContext(ctx context.Context) (res *SolveResult, err
 
 	return ret, err
 }
+
+// WriteLPFile dumps the model to the specified file in lp format.
+func (model *Model) WriteLPFile(filename string) error {
+	model.mu.RLock()
+	defer model.mu.RUnlock()
+
+	ret := C.write_lp(model.prob, C.CString(filename))
+
+	if ret != C.TRUE {
+		return fmt.Errorf("model not written successfully")
+	}
+
+	return nil
+}
+
+//export lpexCallback
+func lpexCallback(modelPtr unsafe.Pointer, buf *C.char) C.int {
+	model, ok := loadRef(modelPtr).(*Model)
+	if !ok {
+		return 0
+	}
+
+	model.writeBuf.WriteString(C.GoString(buf))
+
+	return 0
+}
+
+// WriteLP returns the model in lp format.
+func (model *Model) WriteLP() (string, error) {
+	model.mu.RLock()
+	defer model.mu.RUnlock()
+
+	model.writeBuf.Reset()
```

Comment:
do we really need to keep the buffer in the model? Can't we just instantiate it in the method and use `saveRef`/`loadRef` directly on it?

Reply:
Good point, changed it.